pages/Add/index.js

```diff
@@ -21,33 +21,45 @@ export default function Add({
   setFishLength,
   fishLocation,
   setFishLocation,
+  markerPosition,
+  setMarkerPosition,
 }) {
   const [opened, setOpened] = useState(false);
   const { pathname } = useRouter();
 
   const handleSubmit = (e) => {
     e.preventDefault();
+    {
+      markerPosition.map((marker) => {
```

Comment:
you map over all markers and use the set function in each of them? This will not work since only the last setFistList overwrites all the prior ones.

Reply:
Either you add the fish as soon as a new marker is created or you generate a whole array of new fishes and then append this array to the old fishlist.

Reply:
> Either you add the fish as soon as a new marker is created or you generate a whole array of new fishes and then append this array to the old fishlist.

Sorry, but could you explain a bit more? I tried what i think you meant by that, but it's not working the way i think it should. Maybe i didn't understand your point.

If i delete the .map, then i can't get the lat and lng out of the markerPosition. So i tried to leave it and created a new useState where i append the "newFish". Then the new useState should be merged with the fishList? Can't seem to get this to work.

```js
  const [newFish, setNewFish] = useState(fishList);

  const handleSubmit = (e) => {
    e.preventDefault();

    markerPosition.map((marker) => {
      const newFish = {
        id: nanoid(),
        fishName: fishName,
        fishWeight: +fishWeight,
        fishLength: +fishLength,
        location: fishLocation,
        lat: marker.lat,
        lng: marker.lng,
        date: startDate.toISOString(),
      };

      setNewFish([newFish]);
      setFishList((prevState) => [...prevState, newFish]);
      setOpened(!opened);
      setFishName("");
      setFishWeight("");
      setFishLength("");
      setFishLocation("");
    });
  };

  useEffect(() => {
    setFishList(newFish);
  });
```